src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
so implementations would either unwrap the request and call the underlying client, or the methods of HttpClient would be factory methods to generate the request?

that looks convenient to me. the documentation should state clearly that you should specify PsrHttpClient or HttpClient rather than the combined interface unless you really need both.

about naming: i think we should call Client HttpClient. leaves the question what the thing now called HttpClient should be called? LegacyHttpClient sounds too negative. SimpleHttpClient is a bit weird. its actually a RequestFactoryHttpClient but that name is a bit long.